### PR TITLE
Feature/cdap 13088 queries

### PR DESCRIPTION
**Pull request by CuriousVini**
Reopening https://github.com/caskdata/cdap/pull/10283

### CODE REVIEW

**Comment by CuriousVini** [comment]
@rohitsinha54 Addressed comments from https://github.com/caskdata/cdap/pull/10283 and https://github.com/caskdata/cdap/pull/10324 in the last commit. Please take another pass when you get a chance, thanks!

File: cdap-app-fabric/src/test/java/co/cask/cdap/metadata/MetadataMigratorTest.java [hunk 1]
```diff
@@ -216,46 +205,38 @@ public void testScanAndDelete() throws Exception {
...
       } while (scanCount != 0);
 
-      Assert.assertEquals(13, total);
+      Assert.assertEquals(9, total);
     });
   }
 
   private void assertProperties(MetadataDataset v2System, MetadataDataset v2Business) {
     Assert.assertEquals("avalue11", v2System.getProperties(app1.toMetadataEntity()).get("akey1"));
-    Assert.assertEquals("avalue2", v2System.getProperties(flow1.toMetadataEntity()).get("akey2"));
```
Comment (rohitsinha54) [inline cdap-app-fabric/src/test/java/co/cask/cdap/metadata/MetadataMigratorTest.java:225]: Are we removing this because stream is being deprecated?

File: cdap-app-fabric/src/test/java/co/cask/cdap/metadata/MetadataMigratorTest.java [hunk 2]
```diff
@@ -265,19 +246,18 @@ private void assertIndex(MetadataDataset v2System) throws Exception {
     }
   }
 
-  private void verifyhistory(MetadataDataset v2, MetadataEntity entity, long timestamp) {
+  private void verifyHistory(MetadataDataset v2, MetadataEntity entity, long timestamp) {
     for (Metadata metadata : v2.getSnapshotBeforeTime(ImmutableSet.of(entity), timestamp)) {
-      Assert.assertEquals(1, metadata.getProperties().size());
+      Map<String, String> properties = metadata.getProperties();
```
Comment (rohitsinha54) [inline cdap-app-fabric/src/test/java/co/cask/cdap/metadata/MetadataMigratorTest.java]: why is this change needed?

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MdsHistoryKey.java [hunk 3]
```diff
@@ -23,14 +23,14 @@
...
 import static co.cask.cdap.api.metadata.MetadataEntity.DATASET;
-import static co.cask.cdap.api.metadata.MetadataEntity.DATASET_INSTANCE;
 import static co.cask.cdap.api.metadata.MetadataEntity.PROGRAM;
 import static co.cask.cdap.api.metadata.MetadataEntity.STREAM;
 import static co.cask.cdap.api.metadata.MetadataEntity.VIEW;
 
 /**
  * Key class to get v1 metadata history key information
  */
+@Deprecated
```
Comment (rohitsinha54) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MdsHistoryKey.java]: all deprecated class api should document the reason of deprecation and the release

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MdsKey.java [hunk 4]
```diff
@@ -25,14 +25,14 @@
...
 import static co.cask.cdap.api.metadata.MetadataEntity.DATASET;
-import static co.cask.cdap.api.metadata.MetadataEntity.DATASET_INSTANCE;
 import static co.cask.cdap.api.metadata.MetadataEntity.PROGRAM;
 import static co.cask.cdap.api.metadata.MetadataEntity.STREAM;
 import static co.cask.cdap.api.metadata.MetadataEntity.VIEW;
 
 /**
  * Key class to get v1 metadata key information
  */
+@Deprecated
```
Comment (rohitsinha54) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MdsKey.java]: same as above.

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MdsHistoryKey.java [hunk 5]
```diff
@@ -62,12 +63,13 @@ static long extractTime(byte[] rowKey, String type) {
         break;
       case APPLICATION:
       case DATASET:
```
Comment (rohitsinha54) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MdsHistoryKey.java:66]: if this is only used in migration then it cannot be DATASET_INSTANCE or VIEW as we never serialized that form.

File: cdap-data-fabric/src/test/java/co/cask/cdap/data2/metadata/store/MetadataStoreMigrationTest.java [hunk 6]
```diff
@@ -0,0 +1,243 @@
...
+    txManager = injector.getInstance(TransactionManager.class);
+    txManager.startAndWait();
+
+    store = injector.getInstance(MetadataStore.class);
+    txExecutorFactory = injector.getInstance(TransactionExecutorFactory.class);
+    dsFramework = injector.getInstance(DatasetFramework.class);
+  }
+
+  @AfterClass
+  public static void teardown() {
```
Comment (rohitsinha54) [inline cdap-data-fabric/src/test/java/co/cask/cdap/data2/metadata/store/MetadataStoreMigrationTest.java]: tearDown

**Review by rohitsinha54 — COMMENTED** [review]
Minor comments.

File: cdap-app-fabric/src/test/java/co/cask/cdap/metadata/MetadataMigratorTest.java (hunk 1 above)
Comment (CuriousVini) [reply cdap-app-fabric/src/test/java/co/cask/cdap/metadata/MetadataMigratorTest.java:225]: yes

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MdsHistoryKey.java (hunk 5 above)
Comment (CuriousVini) [reply cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MdsHistoryKey.java:66]: This change is only for extracting timestamp from old metadata key which may have `datasetinstance` or `view` as entity type. When we add same history row in new table, we use new serialization format which does not include `datasetinstance` or `view`.

Comment (rohitsinha54) [reply cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MdsHistoryKey.java:66]: At that point doesn't it through MetadataHistoryKey and not this deprecated class?

**Comment by rohitsinha54** [comment]
One clarification needed. Other than that LGTM
